File: sdk-swift/Sources/DecibelSDK/Models/Vault.swift
import Foundation

public struct Vault: Codable {
    public let address: String
    public let name: String
    public let description: String?
    public let manager: String
    public let status: String
    public let createdAt: Int64
    public let tvl: Double?
    public let volume: Double?
    public let volume30d: Double?
    public let allTimePnl: Double?
    public let netDeposits: Double?
    public let allTimeReturn: Double?
    public let pastMonthReturn: Double?
    public let sharpeRatio: Double?
    public let maxDrawdown: Double?
    public let weeklyWinRate12w: Double?
    public let profitShare: Double?
    public let pnl90d: Double?
    public let managerCashPct: Double?
    public let averageLeverage: Double?
    public let depositors: Int64?
    public let perpEquity: Double?
    public let vaultType: VaultType?
    public let socialLinks: [String]?

    enum CodingKeys: String, CodingKey {
        case address, name, description, manager, status
        case createdAt = "created_at"
        case tvl, volume
        case volume30d = "volume_30d"
        case allTimePnl = "all_time_pnl"
        case netDeposits = "net_deposits"
        case allTimeReturn = "all_time_return"
        case pastMonthReturn = "past_month_return"
        case sharpeRatio = "sharpe_ratio"
        case maxDrawdown = "max_drawdown"
        case weeklyWinRate12w = "weekly_win_rate_12w"
        case profitShare = "profit_share"
        case pnl90d = "pnl_90d"
        case managerCashPct = "manager_cash_pct"
        case averageLeverage = "average_leverage"
        case depositors
        case perpEquity = "perp_equity"
        case vaultType = "vault_type"
        case socialLinks = "social_links"
    }
}

public struct UserOwnedVault: Codable {
    public let vaultAddress: String
    public let vaultName: String
    public let vaultShareSymbol: String
    public let status: String
    public let ageDays: Int64
    public let numManagers: Int64
    public let tvl: Double?
    public let apr: Double?
    public let managerEquity: Double?
    public let managerStake: Double?

    enum CodingKeys: String, CodingKey {
        case vaultAddress = "vault_address"
        case vaultName = "vault_name"
        case vaultShareSymbol = "vault_share_symbol"
        case status
        case ageDays = "age_days"
        case numManagers = "num_managers"
        case tvl, apr
        case managerEquity = "manager_equity"
        case managerStake = "manager_stake"
    }
}

File: sdk-kotlin/src/main/kotlin/trade/decibel/sdk/models/Vault.kt
package trade.decibel.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Vault(
    val address: String,
    val name: String,
    val description: String? = null,
    val manager: String,
    val status: String,
    @SerialName("created_at") val createdAt: Long,
    val tvl: Double? = null,
    val volume: Double? = null,
    @SerialName("volume_30d") val volume30d: Double? = null,
    @SerialName("all_time_pnl") val allTimePnl: Double? = null,
    @SerialName("net_deposits") val netDeposits: Double? = null,
    @SerialName("all_time_return") val allTimeReturn: Double? = null,
    @SerialName("past_month_return") val pastMonthReturn: Double? = null,
    @SerialName("sharpe_ratio") val sharpeRatio: Double? = null,
    @SerialName("max_drawdown") val maxDrawdown: Double? = null,
    @SerialName("weekly_win_rate_12w") val weeklyWinRate12w: Double? = null,
    @SerialName("profit_share") val profitShare: Double? = null,
    @SerialName("pnl_90d") val pnl90d: Double? = null,
    @SerialName("manager_cash_pct") val managerCashPct: Double? = null,
    @SerialName("average_leverage") val averageLeverage: Double? = null,
    val depositors: Long? = null,
    @SerialName("perp_equity") val perpEquity: Double? = null,
    @SerialName("vault_type") val vaultType: String? = null,
    @SerialName("social_links") val socialLinks: List<String>? = null
)

@Serializable
data class UserOwnedVault(
    @SerialName("vault_address") val vaultAddress: String,
    @SerialName("vault_name") val vaultName: String,
    @SerialName("vault_share_symbol") val vaultShareSymbol: String,
    val status: String,
    @SerialName("age_days") val ageDays: Long,
    @SerialName("num_managers") val numManagers: Long,
    val tvl: Double? = null,
    val apr: Double? = null,
    @SerialName("manager_equity") val managerEquity: Double? = null,
    @SerialName("manager_stake") val managerStake: Double? = null
)

@Serializable
data class UserOpenOrder(
    val market: String,
    @SerialName("order_id") val orderId: String,
    @SerialName("client_order_id") val clientOrderId: String? = null,
    val price: Double,
    @SerialName("orig_size") val origSize: Double,
    @SerialName("remaining_size") val remainingSize: Double,
    @SerialName("is_buy") val isBuy: Boolean,
    @SerialName("time_in_force") val timeInForce: String,
    @SerialName("is_reduce_only") val isReduceOnly: Boolean,
    val status: String,
    @SerialName("transaction_unix_ms") val transactionUnixMs: Long,
    @SerialName("transaction_version") val transactionVersion: Long
)
